central/productusage/injector/injector.go

```diff
@@ -0,0 +1,94 @@
+package injector
+
+import (
+	"context"
+	"time"
+
+	datastore "github.com/stackrox/rox/central/productusage/datastore/securedunits"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/sac/resources"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+const period = 1 * time.Hour
+
+var (
+	metricsWriter = sac.AllowFixedScopes(
+		sac.AccessModeScopeKeys(storage.Access_READ_ACCESS, storage.Access_READ_WRITE_ACCESS),
+		sac.ResourceScopeKeys(resources.Administration))
+
+	once     sync.Once
+	injector Injector
+	log      = logging.LoggerForModule()
+)
+
+func (i *injectorImpl) gather(ctx context.Context) {
+	newMetrics, err := i.ds.AggregateAndFlush(ctx)
```

Comment:
I don't really understand why `AggregateAndFlush` and `Upsert` are two different methods and not united

Reply:
Agree, "flush" suffix hints that something is written to a more persistent layer but it's not the case. What the function actually does is `AggregateAndReset()`. Since the in-memory cache is part of the datastore, what is the value of not doing insert in `AggregateAndFlush()`?

Reply:
Will rename in the PR where it is introduced.